### PR TITLE
chore(react-button): adopt custom JSX pragma

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

1. Adopts `react-jsx-runtime` custom pragma on `react-button`

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                     Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `96.657 kB`<br />`23.331 kB` |   `96.61 kB`<br />`23.344 kB` |    `-47 B` <br />`13 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |      `39.845 kB`<br />`10 kB` |  `39.977 kB`<br />`10.032 kB` |    `132 B` <br />`32 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `47.017 kB`<br />`11.469 kB` |  `47.125 kB`<br />`11.508 kB` |    `108 B` <br />`39 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |  `44.532 kB`<br />`11.352 kB` |  `44.663 kB`<br />`11.391 kB` |    `131 B` <br />`39 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |  `53.027 kB`<br />`12.914 kB` |  `53.074 kB`<br />`12.971 kB` |     `47 B` <br />`57 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |  `58.217 kB`<br />`11.922 kB` |  `58.356 kB`<br />`11.961 kB` |    `139 B` <br />`39 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `68.803 kB`<br />`18.728 kB` |  `68.099 kB`<br />`18.467 kB` | `-704 B` <br />`-261 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `208.156 kB`<br />`57.857 kB` | `208.118 kB`<br />`57.879 kB` |    `-38 B` <br />`22 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                  |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                     | `86.164 kB`<br />`24.342 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                               | `80.983 kB`<br />`22.881 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                   |   `9.122 kB`<br />`3.832 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                   |  `11.048 kB`<br />`4.537 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                 |    `9.927 kB`<br />`4.18 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> |  `38.886 kB`<br />`12.41 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                              |   `6.446 kB`<br />`2.185 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/af59ca53d4dfa7e6d80f37002ddcd4107d32b8fa'>af59ca53d4dfa7e6d80f37002ddcd4107d32b8fa</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Button-mount_base.html">384</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Button-mount.html">393</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Button-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Avatar-mount_base.html">713</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Avatar-mount.html">732</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Button-mount_base.html">384</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Button-mount.html">393</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Button-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Field-mount_base.html">1328</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Field-mount.html">1307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/FluentProvider-mount_base.html">893</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/FluentProvider-mount.html">911</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/FluentProviderWithTheme-mount_base.html">109</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/FluentProviderWithTheme-mount.html">118</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">98</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">95</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">114</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">114</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/InfoButton-mount_base.html">21</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/InfoButton-mount.html">18</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/MakeStyles-mount_base.html">1118</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/MakeStyles-mount.html">1128</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Persona-mount_base.html">2040</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/Persona-mount.html">2034</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/SpinButton-mount_base.html">1610</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27605/perf-test-react-components/SpinButton-mount.html">1604</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-button/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "@fluentui/react-tabster": "^9.6.5",
     "@fluentui/react-theme": "^9.1.7",
     "@fluentui/react-utilities": "^9.8.0",
+    "@fluentui/react-jsx-runtime": "9.0.0-alpha.1",
```
Comment (khmakoto) [inline packages/react-components/react-button/package.json:43]: _nit:_ Can you move this above `@fluentui/react-shared-contexts` in line `39` to keep alphabetical order?

**Review by khmakoto — APPROVED** [review]

**Pull request merged**